src/Microsoft.Framework.Configuration.EnvironmentVariables/EnvironmentVariablesConfigurationSource.cs

```diff
@@ -44,7 +44,7 @@ internal void Load(IDictionary envVariables)
                 .ToDictionary(
                     entry => ((string)entry.Key).Substring(_prefix.Length),
                     entry => (string)entry.Value,
-                    StringComparer.OrdinalIgnoreCase);
+                    StringComparer.Ordinal);
```

Comment:
I don't think this behavior is desirable. This makes the dictionary case-sensitive, but config is meant to have case-insensitive keys. I would instead suggest not using the LINQ `.ToDictionary(...)` method and manually iterating over the key/value pairs and adding them safely to a new case-insensitive dictionary. Then the "last key wins" when they are the same except for case.